src/components/DeviceInstructionsModal.tsx

```diff
@@ -1,50 +1,83 @@
 import React from 'react';
 import { Button, Modal, ModalHeader, ModalFooter, ModalBody } from 'reactstrap';
+import { AppState } from '../store';
+import { toggleDeviceInstructionsModal } from '../store/modal'
+import { connect } from 'react-redux';
 
-interface Props {
-	toggle(): any;
-	isOpen: boolean;
+interface DispatchProps {
+  toggleDeviceInstructionsModal: typeof toggleDeviceInstructionsModal;
 }
 
-export default class DeviceInstructionsModal extends React.Component<Props> {
+interface StateProps {
+  open: AppState['modal']['deviceInstructions']['open'];
+  deviceType: AppState['modal']['deviceInstructions']['deviceType'];
+}
+
+type Props = DispatchProps & StateProps
+
+class DeviceInstructionsModal extends React.Component<Props> {
   render() {
 		// TODO: accept an optional "device" prop and only display that device if present 
-		const { toggle, isOpen } = this.props;
-		return (
-			<Modal isOpen={isOpen} toggle={toggle}>
-				<ModalHeader toggle={toggle}>Device Instructions</ModalHeader>
+    const { toggleDeviceInstructionsModal, open, deviceType } = this.props;
```

Comment:
If `deviceType` is present, only instructions for that kind of device are shown. Otherwise, all are shown.